topiary-cli/src/main.rs

```diff
@@ -123,30 +123,41 @@ async fn run() -> CLIResult<()> {
         unreachable!();
     };
 
-    let query_path = if let Some(query) = args.query {
-        query
+    let grammar = language.grammar().await?;
+
+    let query = if let Some(query_path) = args.query.clone() {
+        Ok(query_path)
     } else {
         // Deduce the query file from the language, if the argument is missing
-        language.query_file()?
-    };
-
-    let query_content = (|| {
-        let mut reader = BufReader::new(File::open(&query_path)?);
-        let mut contents = String::new();
-        reader.read_to_string(&mut contents)?;
-
-        Ok(contents)
-    })()
-    .map_err(|e| {
-        TopiaryError::Bin(
-            "Could not open query file".into(),
-            Some(CLIError::IOError(e)),
-        )
+        language.query_file()
+    }
+    .map_err(TopiaryError::Lib)
+    .and_then(|query_path| {
+        {
+            let mut reader = BufReader::new(File::open(query_path)?);
+            let mut contents = String::new();
+            reader.read_to_string(&mut contents)?;
+            Ok(contents)
+        }
+        .map_err(|e| {
+            TopiaryError::Bin(
+                "Could not open query file".into(),
+                Some(CLIError::IOError(e)),
+            )
+        })
+    })
+    .and_then(|query_content: String| Ok(TopiaryQuery::new(&grammar, &query_content)?))
+    .or_else(|e| {
+        // If we weren't able to read the query file, and the user didn't
+        // request a specific query file, we should fall back to the built-in
+        // queries.
```

Comment:
Topiary should log that it is falling back to the builtin query file to avoid future headaches in debugging.

Reply:
Good point!

Reply:
Check now.